pkg/local_workflows/config_utils/feature_flag.go

```diff
@@ -4,27 +4,62 @@ import (
 	"github.com/snyk/go-application-framework/internal/api"
 	"github.com/snyk/go-application-framework/pkg/configuration"
 	"github.com/snyk/go-application-framework/pkg/workflow"
+	"sync"
 )
 
+//go:generate $GOPATH/bin/mockgen -source=feature_flag.go -destination ../../mocks/feature_flag.go -package mocks -self_package github.com/snyk/go-application-framework/pkg/local_workflows/config_utils
+
 func AddFeatureFlagToConfig(engine workflow.Engine, configKey string, featureFlagName string) {
 	config := engine.GetConfiguration()
 
 	callback := func(existingValue interface{}) (interface{}, error) {
 		if existingValue == nil {
-			httpClient := engine.GetNetworkAccess().GetHttpClient()
-			logger := engine.GetLogger()
-			url := config.GetString(configuration.API_URL)
-			org := config.GetString(configuration.ORGANIZATION)
-			apiClient := api.NewApi(url, httpClient)
-			result, err := apiClient.GetFeatureFlag(featureFlagName, org)
-			if err != nil {
-				logger.Printf("Failed to determine feature flag \"%s\" for org \"%s\": %s", featureFlagName, org, err)
-			}
-			return result, nil
+			return CurrentFeatureFlagChecker().GetFeatureFlag(engine, featureFlagName)
 		} else {
 			return existingValue, nil
 		}
 	}
 
 	config.AddDefaultValue(configKey, callback)
 }
+
+type FeatureFlagChecker interface {
+	GetFeatureFlag(engine workflow.Engine, featureFlagName string) (bool, error)
+}
+
+type featureFlagCheckerImpl struct {
+}
+
+var (
+	currentFFC FeatureFlagChecker
```

Comment:
I don't like that we introduce a global variable here. Why is it needed? Can't we just have an instance in the configuration that is created when the configuration is created?